File: src/app/security/app-user-auth.ts
import { UserAuthBase } from "../shared/security/user-auth-base";


export class AppUserAuth extends UserAuthBase {
    CanAccessToDoList: boolean;
    CanEditToDoList: boolean;
    CanDeleteToDoList: boolean;
    CanAddToDoList: boolean;

    init(): void {
        super.init();
        this.CanAccessToDoList = false;
        this.CanEditToDoList = false;
        this.CanDeleteToDoList = false;
        this.CanAddToDoList = false;
    }

    getValueOfProperty(obj: any, key:string):boolean {
        let ret = obj[key];
        return ret;
    }
}

File: ToDoListAPI/ToDoListAPI/EntityClasses/AppUserAuth.cs
namespace ToDoListAPI.EntityClasses
{
    public class AppUserAuth : UserAuthBase
    {
        public AppUserAuth() : base()
        {
            CanAccessToDoList = false;
            CanEditToDoList = false;
            CanAddToDoList = false;
            CanDeleteToDoList = false;
        }

        public bool CanAccessToDoList { get; set; }
        public bool CanEditToDoList { get; set; }
        public bool CanAddToDoList { get; set; }
        public bool CanDeleteToDoList { get; set; }
    }
}
